parsl/providers/lsf/lsf.py

```diff
@@ -0,0 +1,226 @@
+import os
+import time
+import logging
+
+from parsl.channels import LocalChannel
+from parsl.launchers import SingleNodeLauncher
+from parsl.providers.cluster_provider import ClusterProvider
+from parsl.providers.lsf.template import template_string
+from parsl.utils import RepresentationMixin, wtime_to_minutes
+
+logger = logging.getLogger(__name__)
+
+translate_table = {
+    'PEND': 'PENDING',
+    'RUN': 'RUNNING',
+    'DONE': 'COMPLETED',
+    'EXIT': 'FAILED',  # (failed),
+    'PSUSP': 'CANCELLED',
+    'USUSP': 'CANCELLED',
+    'SSUSP': 'CANCELLED',
+}  # (special exit state
+
+
+class LSFProvider(ClusterProvider, RepresentationMixin):
+    """Slurm Execution Provider
+
+    This provider uses sbatch to submit, squeue for status and scancel to cancel
+    jobs. The sbatch script to be used is created from a template file in this
+    same module.
+
+    Parameters
+    ----------
+    channel : Channel
+        Channel for accessing this provider. Possible channels include
+        :class:`~parsl.channels.LocalChannel` (the default),
+        :class:`~parsl.channels.SSHChannel`, or
+        :class:`~parsl.channels.SSHInteractiveLoginChannel`.
+    nodes_per_block : int
+        Nodes to provision per block.
+    min_blocks : int
+        Minimum number of blocks to maintain.
+    max_blocks : int
+        Maximum number of blocks to maintain.
+    parallelism : float
+        Ratio of provisioned task slots to active tasks. A parallelism value of 1 represents aggressive
+        scaling where as many resources as possible are used; parallelism close to 0 represents
+        the opposite situation in which as few resources as possible (i.e., min_blocks) are used.
+    walltime : str
+        Walltime requested per block in HH:MM:SS.
+    project : str
+        Project to which the resources must be charged
+    scheduler_options : str
+        String to prepend to the #SBATCH blocks in the submit script to the scheduler.
+    worker_init : str
+        Command to be run before starting a worker, such as 'module load Anaconda; source activate env'.
+    launcher : Launcher
+        Launcher for this provider. Possible launchers include
+        :class:`~parsl.launchers.SingleNodeLauncher` (the default),
+        :class:`~parsl.launchers.SrunLauncher`, or
+        :class:`~parsl.launchers.AprunLauncher`
+     move_files : Optional[Bool]: should files be moved? by default, Parsl will try to move files.
+    """
+
+    def __init__(self,
+                 channel=LocalChannel(),
+                 nodes_per_block=1,
+                 init_blocks=1,
+                 min_blocks=0,
+                 max_blocks=10,
+                 parallelism=1,
+                 walltime="00:10:00",
+                 scheduler_options='',
+                 worker_init='',
+                 project=None,
+                 cmd_timeout=10,
```

Comment:
This is not documented in the docstring.

I think this 10s default is too low for real use on large systems? some of our provides default to 120s (pbspro, torque)

Reply:
Reasonable. I've updated this.